src/sagemaker/jumpstart/session_utils.py

```diff
@@ -0,0 +1,159 @@
+        )
+    return model_id, model_version, inference_component_name
+
+
+def _get_model_id_version_from_ic_endpoint_without_ic_name(
+    endpoint_name: str, sagemaker_session: Session
+) -> Tuple[str, str, str]:
+    """Given an endpoint name, derives the model ID, version, and inferred inference component name.
+
+    This function assumes the endpoint corresponds to an inference component-based endpoint.
+
+    Raises:
+        ValueError: If there is not a single inference component associated with the endpoint.
+    """
+    inference_components = sagemaker_session.list_inference_components(
+        endpoint_name_equals=endpoint_name
+    )["InferenceComponents"]
+
+    if len(inference_components) == 0:
+        raise ValueError(
+            f"No inference components found for the following endpoint: {endpoint_name}. "
+            "Use ``SageMaker.CreateInferenceComponent`` to add inference components to "
+            "your endpoint."
+        )
+    if len(inference_components) > 1:
+        raise ValueError(
+            "Must provide inference component name for endpoint with "
+            "multiple inference components."
+        )
+    inference_component_name = inference_components[0]["InferenceComponentName"]
+    return (
+        *_get_model_id_version_from_ic_endpoint_with_ic_name(
+            inference_component_name, sagemaker_session
+        ),
+        inference_component_name,
+    )
+
+
+def _get_model_id_version_from_ic_endpoint_with_ic_name(
+    inference_component_name: str, sagemaker_session: Session
+):
+    """Returns the model ID and version inferred from a sagemaker inference component.
+
+    Raises:
+        ValueError: If the inference component does not have tags from which the model ID
+            and version can be inferred.
+    """
+    region: str = sagemaker_session.boto_region_name
+    partition: str = aws_partition(region)
+    account_id: str = sagemaker_session.account_id()
+
+    inference_component_arn = (
+        f"arn:{partition}:sagemaker:{region}:{account_id}:"
+        f"inference-component/{inference_component_name}"
+    )
+
+    model_id, model_version = get_jumpstart_model_id_version_from_resource_arn(
+        inference_component_arn, sagemaker_session
+    )
+
+    if not model_id:
+        raise ValueError(
+            "Cannot infer JumpStart model ID from inference component "
+            f"'{inference_component_name}'. Please specify JumpStart `model_id` "
+            "when retrieving default predictor for this inference component."
+        )
+
+    return model_id, model_version
+
+
+def _get_model_id_version_from_non_ic_endpoint(
+    endpoint_name: str,
+    inference_component_name: Optional[str],
+    sagemaker_session: Session,
+) -> Tuple[str, str]:
+    """Returns the model ID and version inferred from a model-based endpoint..
+
+    Raises:
+        ValueError: If a non-None inference component name is supplied, or if the endpoint does
```

Comment:
Could you review docstring with doc team? `non-None inference component name` is technically correct; sounds a little uncommon but cute to me. hh

Reply:
so...this method exposes a parameter, but if you pass anything to it, it raises a `ValueError`???

Was that your intent?!

Reply:
Yeah, this is a private method, so it has very particular behavior.